src/wolfsftp.c

```diff
@@ -2301,27 +2268,55 @@ int wolfSSH_SFTP_RecvOpenDir(WOLFSSH* ssh, int reqId, byte* data, word32 maxSz)
     }
     WMEMCPY(dirName, data + idx, sz);
     dirName[sz] = '\0';
-    if (wolfSSH_CleanPath(ssh, dirName) < 0) {
-        WFREE(dirName, ssh->ctx->heap, DYNTYPE_BUFFER);
-        return WS_FATAL_ERROR;
+
+    /* Special case in Windows for the root directory above the drives. */
+    if (dirName[0] == '/' && dirName[1] == 0) {
+        DWORD drives, mask;
+        UINT driveType;
+        char driveName[] = " :\\";
+        int i;
+
+        WMEMSET(ssh->driveList, 0, sizeof ssh->driveList);
+        ssh->driveListCount = 0;
+
+        drives = GetLogicalDrives();
+        for (i = 0, mask = 1; i < (sizeof ssh->driveList); i++, mask <<= 1) {
+            if (drives & mask) {
+                driveName[0] = 'A' + i;
+                driveType = GetDriveTypeA(driveName);
+                if (driveType == DRIVE_FIXED || driveType == DRIVE_REMOTE) {
+                    ssh->driveList[ssh->driveListCount++] = driveName[0];
+                }
+            }
+        }
+        ssh->driveIdx = 0;
     }
+    else {
+        if (sz > MAX_PATH - 2) {
+            WLOG(WS_LOG_SFTP, "Path name is too long.");
+            return WS_FATAL_ERROR;
+        }
+        WSTRNCPY(name, dirName, MAX_PATH);
+        WSTRNCAT(name, "/*", MAX_PATH);
 
-    /* get directory handle */
-    findHandle = (HANDLE)WS_FindFirstFileA(dirName,
-            realName, sizeof(realName), &isDir, ssh->ctx->heap);
-    if (findHandle == INVALID_HANDLE_VALUE || !isDir) {
+        /* get directory handle - see if directory exists */
+        findHandle = (HANDLE)WS_FindFirstFileA(name,
+                realName, sizeof(realName), &isDir, ssh->ctx->heap);
+        if (findHandle == INVALID_HANDLE_VALUE || !isDir) {
 
-        WLOG(WS_LOG_SFTP, "Error with opening directory");
-        WFREE(dirName, ssh->ctx->heap, DYNTYPE_BUFFER);
+            WLOG(WS_LOG_SFTP, "Error with opening directory");
```

Comment:
Would be helpful to add the directory path to this log.

Reply:
Did both instances of this error.